### PR TITLE
Python 3.9 - Buster

### LINKED ISSUE #387: Upgrade to Debian buster
The docker images still use stretch.

### PR DESCRIPTION
**Pull request by ibot3**
This PR changes the python version to 3.9 based on debian buster.

Closes #387

### CODE REVIEW

**Review by georgeto — APPROVED** [review]
Looks good to me

File: tests/lib/test_finance.py [hunk 1]
```diff
@@ -605,17 +605,16 @@ class MatchingTestCase(unittest.TestCase):
         ("12345-65, Hans Wurst, HSS46/A 01 B", None),  # bad checksum
         ("12345-20, Hans Wurst, HSS46/A 01 B", "pyc-12345"),
         ("1, Hans Wurst, HSS46/A 01 B", None),
-        ("n1colas, Nicolas Bourbaki, HSS48 76-3", "hss-n1colas"),
-        ("n1cOLAS, Nicolas Bourbaki, HSS48 76-3", "hss-n1colas"),
-        ("  admin,Ich Bin Hier der Admin , ficticous location , garbage", "hss-admin"),
-        ("FOO, FOO BAR, HSS46 16-11", "hss-foo"),
```
Comment (lukasjuhrich) [inline tests/lib/test_finance.py:611]: How are these changes related to the PR?

File: pycroft/helpers/i18n.py [hunk 2]
```diff
@@ -337,7 +337,7 @@ class Message(object):
     @classmethod
     def from_json(cls, json_string):
         try:
-            obj = json.loads(json_string, encoding='utf8')
+            obj = json.loads(json_string)
```
Comment (lukasjuhrich) [inline pycroft/helpers/i18n.py:340]: How is this related to the PR? In both cases this should be the py3.7  stdlib `json`-module, so there should'nt be any breaking API changes

Comment (ibot3) [reply pycroft/helpers/i18n.py:340]: As written in the commit description, the `encoding` parameter has been deprecated since 3.1 and was removed with 3.9.

File: tests/lib/test_finance.py (hunk 1 above)
Comment (ibot3) [reply tests/lib/test_finance.py:611]: Not directly, I just wanted to make sure that all tests are running fine.

File: pycroft/helpers/i18n.py (hunk 2 above)
Comment (lukasjuhrich) [reply pycroft/helpers/i18n.py:340]: Thanks for the clarification, didn't see the corresponding commit message. Fine by me

**Comment by lukasjuhrich** [comment]
Assigning myself this PR so I can rebase && push after #427 goes through.
